docs/recipes/build/rollup.md

```diff
@@ -0,0 +1,144 @@
+# Rollup-based Bundling
+
+These recipes cover the following `sw-lib` use case:
+- You want to write and maintain your own `sw.js` file that uses precaching via
+`sw-lib.cacheRevisionedAssets()` alongside other service worker logic.
+- You'd like to use [ES2015 import syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import)
+to include `sw-lib` and other modules into your service worker.
+- You are comfortable using [`Rollup`](https://github.com/rollup/rollup) to
+handle transforming the `sw.js` file you maintain into a production-ready,
+self contained service worker file, containing the `sw-lib` library code along
+with your custom logic.
+- You are comfortable using either [`gulp`](http://gulpjs.com/) or
+[`npm scripts`](https://docs.npmjs.com/misc/scripts) as a task runner for your
+build process.
+
+## Dependencies
+
+- Working installations of `node` (4.0+) & `npm`.
+- `npm install --save-dev sw-build sw-lib rollup rollup-plugin-node-resolve`
+
+## Service Worker Code
+
+This approach assumes that you have a `src/sw.js` that uses [ES2015 module
+import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import)
+syntax, and is structured like:
+
+```js
+// Contents of src/sw.js:
+
+import manifest from '/tmp/manifest.js';
+import swLib from 'sw-lib';
+// imports for any other libraries, e.g. idb-keyval, Firebase Messaging, etc.
+
+// This will precache the files in your manifest, and keep them up to date.
+swLib.cacheRevisionedAssets(manifest);
+
+// Any additional runtime service worker logic can go here.
+```
+
+## Your Build Process
+
+Here are two alternatives that lead to an equivalent, ready-to-deploy service
+worker file. You can choose between them based on whether you'd prefer using
+`npm scripts` or `gulp` as your task runner.
+
+**Note:** Regardless of which task runner you're using, make sure that the
+generation of the service worker file happens *last* in the chain of your build
+tasks. This ensures that the service worker looks at the final contents of your
+`build/` directory when determining which files to precache and keep up to date.
+
+### `npm scripts`
+
+The following is a basic `node` script that can be run as part of a larger build
```

Comment:
If you're following this recipe, it's not like to look like `"scripts": {"build": "node ./build.js"}`

More likely is `"scripts": {"build": "node build.js && node build-sw.js"}` (chaining the additional build process outlined here to the end of your existing build process).

But I can include that sample.